src/vfio/core.c

```diff
@@ -215,9 +310,37 @@ static int vfio_do_map_dma(struct vfio_container *vfio, void *vaddr, size_t len,
 	}
 
 	return 0;
+
 }
 
-static int vfio_do_unmap_dma(struct vfio_container *vfio, size_t len, uint64_t iova)
+static int vfio_iommufd_do_map_dma(struct vfio_container *vfio,
```

Comment:
Sorry if I'm beating a dead horse here, but to the refactor idea: We have `vfio_container_do_map_dma` and ` vfio_iommufd_do_map_dma` that are both taking a `struct vfio_container *vfio`.

I would propose something like: `vfio_do_map_dma` and `iommufd_do_map_dma`. Either they take their respective `vfio_container` and `iommufd_...` or we come up with a generic interface.

Reply:
I think it would be something like "do_map_dma" and the backend implementation would be selected at compile time.

As for the struct passed to the do_map_dma functions. I think we can just do away with the struct and pass an fd as that is the only thing that I see being used.